app/src/org/commcare/activities/RefreshToLatestBuildActivity.java

```diff
@@ -0,0 +1,122 @@
+package org.commcare.activities;
+
+import android.app.Activity;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import org.commcare.CommCareApplication;
+import org.commcare.dalvik.R;
+import org.commcare.preferences.DevSessionRestorer;
+import org.commcare.preferences.DeveloperPreferences;
+import org.commcare.utils.SessionUnavailableException;
+import org.commcare.views.dialogs.AlertDialogFactory;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Activity that is launched immediately upon reception of a RefreshToLatestBuildAction broadcast,
+ * triggering the necessary action sequence.
```

Comment:
This comment talks about how this class is used in our codebase, but doesn't describe what it does. I would find it helpful to include what this activity is doing / showing to the user.

I've read arguments that adding comments describing how code is used is harmful because as the codebase changes most developers neglect to keep these comments update-to-date. I tend to agree with this idea.